nixos/lib/test-driver/test-driver.py

```diff
@@ -973,7 +973,7 @@ def subtest(name: str) -> Iterator[None]:
 
 
 if __name__ == "__main__":
-    arg_parser = argparse.ArgumentParser()
+    arg_parser = argparse.ArgumentParser(prog="@name@")
```

Comment:
i stumble on this one: it appears evitable to do raw text-sustitution-preprocessing on the test script. if we start doing this, then we could/should substitute a lot of other things, too.

Reply:
ok it's too early for me, it is about substitution of course.

but i don't understand why we need to mix such things again. initially, you wanted to refactor the driver a bit to prepare for the upcoming tmux changes. now i am reviewing all kinds of different things that i wasn't expecting and this with all the coupled discussions drags too much time. i won't be able to timely accompany such efforts if we keep mixing things up.

Reply:
No worries! :wink:

This substitution, as the comment in `testing-python.nix` suggests is because 
`exec -a` doesn't seem to have effect for python shebangs.

So if you run the script and get the help text, then it would show as `.nixos-test-driver-wrapper` which does not correspond to what you did and hence is confusing ("a lie").